test/common/unit.js

```diff
@@ -197,6 +197,11 @@ describe('Unit', () => {
     t.columns.add('c', sql.TinyInt, { nullable: false, primary: true })
     assert.strictEqual(t.declare(), 'create table [#mytemptable] ([a] int, [b] tinyint null, [c] tinyint not null, constraint PK_mytemptable primary key (a, c))')
 
+    t = new sql.Table('MyTable')
+    t.columns.add('name', sql.NVarChar, {
+      length: Infinity
+    })
+    assert.strictEqual(t.columns[0].length, Infinity)
```

Comment:
How do falsey (ie. null/undefined) and normal numbers behave here?

Reply:
https://tediousjs.github.io/tedious/bulk-load.html
As per tedious documentation, `length for VarChar, NVarChar, VarBinary. Use length as Infinity for VarChar(max), NVarChar(max) and VarBinary(max)`.
Using number datatype for length field (including Infinity which is also of type ‘number’ ) will assign it to the value of the column, if we specify anything other than a number like undefined, string or null 
It will throw “ `UnhandledPromiseRejectionWarning: RequestError: An unknown error has occurred. This is likely because the schema of the BulkLoad does not match the schema of the table you are attempting to insert into`.” error.

Reply:
https://github.com/tediousjs/tedious/issues/679 - Tedious issue for the same error

Reply:
OK, so how does `length: 'max'` or `length: null` or `length: ''` behave? I'm asking to see tests for those values.

Reply:
@willmorgan hi will , added test case for the above scenarios as well, length 'max' works , but any other string value will throw `Incorrect syntax near '' error`, for undefined values it will throw `Invalid column type from bcp client for colid 1.`. please review and suggest any more changes needed.